### PR TITLE
SDK generation from 2021-12-01 Compute Swagger

### PR DESCRIPTION
**Pull request by haagha**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/DiskUpdate.cs [hunk 1]
```diff
@@ -79,15 +79,17 @@ public DiskUpdate()
...
+        public DiskUpdate(OperatingSystemTypes? osType = default(OperatingSystemTypes?), int? diskSizeGB = default(int?), EncryptionSettingsCollection encryptionSettingsCollection = default(EncryptionSettingsCollection), long? diskIOPSReadWrite = default(long?), long? diskMBpsReadWrite = default(long?), long? diskIOPSReadOnly = default(long?), long? diskMBpsReadOnly = default(long?), int? maxShares = default(int?), Encryption encryption = default(Encryption), string networkAccessPolicy = default(string), string diskAccessId = default(string), string tier = default(string), bool? burstingEnabled = default(bool?), PurchasePlan purchasePlan = default(PurchasePlan), SupportedCapabilities supportedCapabilities = default(SupportedCapabilities), PropertyUpdatesInProgress propertyUpdatesInProgress = default(PropertyUpdatesInProgress), bool? supportsHibernation = default(bool?), string publicNetworkAccess = default(string), string dataAccessAuthMode = default(string), IDictionary<string, string> tags = default(IDictionary<string, string>), DiskSku sku = default(DiskSku))
         {
```
Comment (haagha) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/DiskUpdate.cs:93]: add customization

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/SnapshotUpdate.cs [hunk 2]
```diff
@@ -56,11 +56,12 @@ public SnapshotUpdate()
...
+        public SnapshotUpdate(OperatingSystemTypes? osType = default(OperatingSystemTypes?), int? diskSizeGB = default(int?), EncryptionSettingsCollection encryptionSettingsCollection = default(EncryptionSettingsCollection), Encryption encryption = default(Encryption), string networkAccessPolicy = default(string), string diskAccessId = default(string), bool? supportsHibernation = default(bool?), string publicNetworkAccess = default(string), string dataAccessAuthMode = default(string), SupportedCapabilities supportedCapabilities = default(SupportedCapabilities), IDictionary<string, string> tags = default(IDictionary<string, string>), SnapshotSku sku = default(SnapshotSku))
         {
```
Comment (haagha) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/SnapshotUpdate.cs:65]: customization

Comment (haagha) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/SnapshotUpdate.cs:65]: added

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/DiskUpdate.cs (hunk 1 above)
Comment (haagha) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/DiskUpdate.cs:93]: added

**Comment by haagha** [comment]
@archerzz The CI checks are failing because of Azure.ResourceManager.Compute, which this PR does not touch. can you help?

**Comment by grizzlytheodore** [comment]
ready for merging. Please review merge!  @archerzz @markcowl

**Comment by grizzlytheodore** [comment]
please help merge @ArthurMa1978

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**